src/CacheInterface.php

```diff
@@ -2,12 +2,22 @@
 
 namespace React\Cache;
 
+use React\Promise\PromiseInterface;
+
 interface CacheInterface
 {
-    // @return React\Promise\PromiseInterface
+    /**
+     * @return PromiseInterface
+     */
     public function get($key);
 
+    /**
+     * @return PromiseInterface
+     */
     public function set($key, $value);
 
+    /**
+     * @return PromiseInterface
```

Comment:
@jsor exactly. By defining `@return PromiseInterface|mixed` the IDE will support code completion by using `PromiseInterface` and developer will use second part. If implementing library want to narrow it to `PromiseInterface|string`, then developer will know that fulfilled `PromiseInterface` will result with `string` type variable.